### PR TITLE
Csv downloader

### PR DESCRIPTION
**Pull request by BenjaminCall**
@BenHSmith13 or @dittonjs When you have some time look this over please.

### CODE REVIEW

File: app/controllers/download_status_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class DownloadStatusController < ApplicationController
+  include Concerns::JwtToken
+  before_action :validate_token
+
+  def index
+    exams_list = AssignedExam.where(course_id: params[:course_id], exam_id: params[:exam_id])
+
+    final_countdown_csv = CSV.generate do |csv|
```
Comment (BenHSmith13) [inline app/controllers/download_status_controller.rb]: Do do do, do do do do
https://www.youtube.com/watch?v=9jK-NcRmVcw

File: client/js/proctor/components/exam_distribution/_exam_distribution.jsx [hunk 2]
```diff
@@ -199,10 +202,21 @@ export class BaseExamDistribution extends React.Component {
...
     return (
       <div>
         <h1 style={styles.header}>
           {this.props.exam.title}
+          <HoverButton
+            className="spec_download"
+            style={styles.floatRight}
+            onClick={() => this.props.downloadExamStatus(params.id, lmsCourseId)}
+          >
+            <svg xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
```
Comment (BenHSmith13) [inline client/js/proctor/components/exam_distribution/_exam_distribution.jsx]: put this fella in his own component

File: client/js/proctor/components/exam_distribution/student_assign.jsx [hunk 3]
```diff
@@ -83,14 +83,17 @@ export default class StudentAssign extends React.Component {
...
-
     let assignObject;
     if (this.state.selectedCenterId != null && !assignedExam) {
       assignObject = (
         <HoverButton
           style={styles.button}
           hoveredStyle={styles.hoveredStyle}
-          onClick={() => this.props.assignExam(this.props.student, this.state.selectedCenterId)}
+          onClick={() => this.props.assignExam(
+            this.props.student,
```
Comment (BenHSmith13) [inline client/js/proctor/components/exam_distribution/student_assign.jsx]: throw these prop values into the prop deconstructor.

File: client/js/proctor/components/exam_distribution/student_assign.jsx [hunk 4]
```diff
@@ -102,7 +105,11 @@ export default class StudentAssign extends React.Component {
         <HoverButton
           style={styles.button}
           hoveredStyle={styles.hoveredStyle}
-          onClick={() => this.reassignExam(this.props.assignedExam.id, this.state.selectedCenterId)}
+          onClick={() => this.reassignExam(
+            this.props.assignedExam.id,
```
Comment (BenHSmith13) [inline client/js/proctor/components/exam_distribution/student_assign.jsx]: ditto

File: config/routes.rb [hunk 5]
```diff
@@ -37,6 +37,7 @@ def matches?(request)
   end
 
   resources :users
+  resources :download_status
```
Comment (BenjaminCall) [inline config/routes.rb]: only allow the index route with get

**Comment by nick-benoit14** [comment]
Looks good to me!

File: client/js/proctor/components/exam_distribution/_exam_distribution.jsx [hunk 6]
```diff
@@ -199,10 +211,18 @@ export class BaseExamDistribution extends React.Component {
 
   render() {
     const styles = BaseExamDistribution.getStyles();
+    const { params, lmsCourseId } = this.props;
```
Comment (BenjaminCall) [inline client/js/proctor/components/exam_distribution/_exam_distribution.jsx]: downloadExam props destruct

File: client/js/proctor/components/exam_distribution/_exam_distribution.jsx [hunk 7]
```diff
@@ -199,10 +211,18 @@ export class BaseExamDistribution extends React.Component {
...
     return (
       <div>
         <h1 style={styles.header}>
           {this.props.exam.title}
+          <HoverButton
+            className="spec_download"
+            style={styles.downloadSvg}
+            onClick={() => this.props.downloadExamStatus(params.id, lmsCourseId)}
+          >
+            <DownLoadSvg />
```
Comment (BenjaminCall) [inline client/js/proctor/components/exam_distribution/_exam_distribution.jsx]: DownLoad??... Download

File: client/js/proctor/components/exam_distribution/student_assign.jsx [hunk 8]
```diff
@@ -82,15 +82,20 @@ export default class StudentAssign extends React.Component {
 
   render() {
     const styles = StudentAssign.getStyles();
-    const { assignedExam } = this.props;
-
+    const { assignedExam, testingCenterList, student } = this.props;
     let assignObject;
+    const selectedCenterId = this.state.selectedCenterId;
```
Comment (BenjaminCall) [inline client/js/proctor/components/exam_distribution/student_assign.jsx]: destruct better

**Comment by BenHSmith13** [comment]
Looks good

**Pull request merged**